appinventor/components-ios/src/BarChartDataModel.swift

```diff
@@ -62,8 +77,18 @@ open class BarChartDataModel: Chart2DDataModel {
       let flooredX = Int(floor(x))
       return BarChartDataEntry(x: Double(flooredX), y: y)
     } else {
-      // Handle number format exception
-      return nil
+      do {
+        guard let dateX = formatDateTime(rawX) else {
+            throw ChartDataError.invalidDateString(rawX)
+        }
+        guard let y = Double(rawY) else {
+            throw ChartDataError.invalidValue(rawY)
+        }
+        return BarChartDataEntry(x: dateX, y: y)
+      } catch {
+        print("Error parsing x or y coordinates: \(rawX), \(rawY), error:\(error)")
```

Comment:
It looks like this is just writing an error to the log. Do we want to throw an error here?

Reply:
> I wasn't sure. I guess since the java correlative does it, I should

Yes, I think so.